### DISCUSSION TITLE
Is it possible to debug pudb/urwid itself?

### DISCUSSION THREAD

**Discussion by marc-h38** [inducer/pudb#622] [Q&A]
I'm trying to debug https://github.com/inducer/pudb/issues/598 and I'm wondering whether it's possible to debug pudb _itself_?

I tried the patch below + `./try-the-debugger.sh` and I'm getting strange behaviors:
- I get the warning below
- pdb seems to "override" pudb but not really
- reproduction of 598 seems semi-randomly affected

```
pudb/pudb/__init__.py:293: UserWarning: A non-default handler for signal 2 is already installed (<bound method Pdb.sigint_handler of <pdb.Pdb object at 0x7fe120dbb5d0>>). Skipping pudb interrupt support.
  return warn("A non-default handler for signal %d is already installed (%s). "
```

```diff
--- a/urwid/raw_display.py
+++ b/urwid/raw_display.py
         if self.screen_buf and r is self._screen_buf_canvas:
             return
 
         self._setup_G1()
+       import pdb; pdb.set_trace()
 
         if self._resized:
             # handle resize before trying to draw screen
```

TBH I'm not too surprised this does not really work; I just want to know whether this is theoritically possible? I mean should I persist or give up or try something else?

**Comment by inducer** [comment]
I've never even tried; so I don't know. Hypothetically, any of the 'remote debugging' options might work, though I don't know how well two `Bdb` instances (the underlying Python debugging infrastructure) get along. In short, I really don't know.

**Reply by marc-h38** [reply]
"The shoemaker's children go barefoot"
